### PR TITLE
feat: test workflow execution model

### PR DESCRIPTION
**Pull request by vsukhin**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [x] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: api/testworkflows/v1/testworkflowexecution_type.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+	StatusAt metav1.Time `json:"statusAt,omitempty" expr:"template"`
+	// structured tree of steps
+	Signature []TestWorkflowSignature `json:"signature,omitempty" expr:"include"`
+	Result    *TestWorkflowResult     `json:"result,omitempty" expr:"include"`
+	// additional information from the steps, like referenced executed tests or artifacts
+	Output []TestWorkflowOutput `json:"output,omitempty" expr:"include"`
+	// generated reports from the steps, like junit
+	Reports          []TestWorkflowReport `json:"reports,omitempty" expr:"include"`
+	Workflow         *TestWorkflow        `json:"workflow" expr:"include"`
+	ResolvedWorkflow *TestWorkflow        `json:"resolvedWorkflow,omitempty" expr:"include"`
```
Comment (rangoo94) [inline api/testworkflows/v1/testworkflowexecution_type.go:71]: There is a chance that you may need to mark them as schemaless - the TestWorkflow CRD is near Kubernetes constraints, while there are two of them; Unfortunately, Kubernetes doesn't support `$ref` yet 🙁

File: api/testworkflows/v1/testworkflowexecution_type.go [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+	// test workflow execution name started the test workflow execution
+	TestWorkflowExecutionName string `json:"testWorkflowExecutionName,omitempty" expr:"template"`
+}
+
+// TestWorkflowSignature has signature of TestWorkflow
+type TestWorkflowSignature struct {
```
Comment (rangoo94) [inline api/testworkflows/v1/testworkflowexecution_type.go:77]: The values in signature are fully computed, so rather `expr:"ignore"` for each of them

File: api/testworkflows/v1/testworkflowexecution_type.go [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+// TestWorkflowExecutionDetails contains the details of TestWorkflowExecution
+type TestWorkflowExecutionDetails struct {
+	// unique execution identifier
+	Id string `json:"id" expr:"template"`
+	// execution name
+	Name string `json:"name" expr:"template"`
+	// execution namespace
+	Namespace string `json:"namespace,omitempty" expr:"template"`
+	// sequence number for the execution
+	Number int32 `json:"number,omitempty" expr:"template"`
```
Comment (rangoo94) [inline api/testworkflows/v1/testworkflowexecution_type.go:58]: `expr:"ignore"`, as it's bare integer

File: api/testworkflows/v1/testworkflowexecution_type.go [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+	TestWorkflowExecutionName string `json:"testWorkflowExecutionName,omitempty" expr:"template"`
+}
+
+// TestWorkflowExecutionStatus defines the observed state of TestWorkflowExecution
+type TestWorkflowExecutionStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	LatestExecution *TestWorkflowExecutionDetails `json:"latestExecution,omitempty" expr:"include"`
+	// test workflow execution generation
+	Generation int64 `json:"generation,omitempty" expr:"template"`
```
Comment (rangoo94) [inline api/testworkflows/v1/testworkflowexecution_type.go:46]: `expr:"ignore"`

**Review by rangoo94 — APPROVED** [review]
Generally worth to revisit the `expr:` tags in these objects (as for sure most of them should not be there or `expr:"ignore"` should be used), as the objects that are already computed shouldn't have it.

It shouldn't break anything yet though (except maybe doubled TestWorkflow without schemaless, but to check with `make install`), so approving 👍

**Comment by vsukhin** [comment]
yeah, agree, thank you will update them

File: api/testworkflows/v1/testworkflowexecution_type.go (hunk 2 above)
Comment (vsukhin) [reply api/testworkflows/v1/testworkflowexecution_type.go:81]: removed

File: api/testworkflows/v1/testworkflowexecution_type.go (hunk 3 above)
Comment (vsukhin) [reply api/testworkflows/v1/testworkflowexecution_type.go]: removed

File: api/testworkflows/v1/testworkflowexecution_type.go (hunk 4 above)
Comment (vsukhin) [reply api/testworkflows/v1/testworkflowexecution_type.go]: removed

**Pull request merged**